packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -137,6 +137,15 @@ sealed class _DebugSemanticsRoleChecks {
 
   static FlutterError? _noCheckRequired(SemanticsNode node) => null;
 
+  static FlutterError? _semanticsComboBox(SemanticsNode node) {
+    final SemanticsData data = node.getSemanticsData();
+    if (!data.hasFlag(SemanticsFlag.hasExpandedState)) {
+      return FlutterError('A combo box needs expanded states');
```

Comment:
also check if it is expanded, it should also have children

Reply:
Also we should add a test in flutter/packages/flutter/test/widgets/semantics_role_checks_test.dart